src/Header/FileOperator.h

```diff
@@ -2,40 +2,39 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-#ifndef GROUP4_SRC_FILEOPERATOR_H_
-#define GROUP4_SRC_FILEOPERATOR_H_
+#ifndef GROUP4_SRC_HEADER_FILEOPERATOR_H_
+#define GROUP4_SRC_HEADER_FILEOPERATOR_H_
 
 #include <string>
-
 #include "OperateData.h"
 #include "ResponseHandler.h"
-#include "StreamConnector.h"
-
-// File management for client
-class FileOperator {
- public:
-  // Constructor
-  FileOperator();
-
-  // Upload file to server.
-  void Upload(const std::string &path, const std::string &filename,
-              const std::string &host, const int &port);
+#include "APIHandler.h"
 
-  // Down file to client.
-  void Download(const std::string &path, const std::string &host,
-                const int &port);
 
-  // Update file info. For example, file name or owner, file authority.
-  void Update(const std::string &path, const &OperatorData updateInfo,
-              const std::string &host, const int &port);
-
-  // Not copyable or movable
-  FileOperator(const FileOperator &copy) = delete;
-  FileOperator &operator=(const FileOperator &copy) = delete;
-
- private:
-  // Use ResponseHandler to call corresponding service
-  ResponseHandler response;
+// File management for client
+class FileOperator : public APIHandler
+{
+public:
+    // Constructor
+    FileOperator();
+    
+    // Upload file to server.
+    int Upload(const std::unique_ptr<OperateData> &data);
```

Comment:
Usually we pass std::unique_ptr in without the const or reference, because we're passing the ownership in with unique_ptr.

Please change this to without const and without reference.
Note that callers of this function should use std::move() to pass the unique_ptr in.